flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/reader/split/PulsarOrderedPartitionSplitReader.java

```diff
@@ -91,8 +91,15 @@ protected void startConsumer(PulsarPartitionSplit split, Consumer<byte[]> consum
                     initialPosition = nextMessageId(latestConsumedId);
                 }
 
-                consumer.seek(initialPosition);
-            } catch (PulsarClientException e) {
+                // Remove Consumer.seek() here for waiting for pulsar-client-all 2.12.0
+                // See https://github.com/apache/pulsar/issues/16757 for more details.
+                pulsarAdmin
+                        .topics()
+                        .resetCursor(
+                                split.getPartition().getFullTopicName(),
+                                sourceConfiguration.getSubscriptionName(),
+                                initialPosition);
+            } catch (PulsarAdminException e) {
```

Comment:
Cool.

It seems CI failed on Kafka tests. But you may retrigger it and we can merge this one once CI gives green.